cpp/src/io/csv/csv_reader.cu

```diff
@@ -437,37 +519,25 @@ gdf_error read_csv(csv_read_arg *args)
 
 	// Check if the user gave us a list of column names
 	if(args->names == nullptr) {
-		int h_num_cols = 0;
-
-		if (raw_csv->header_row >= 0) {
-			// Parse header row to assign column names; does not currently handle
-			// quotations to avoid added parsing complexity
-			for (size_t pos = 0, prev = 0; pos < raw_csv->header.size(); ++pos) {
-				if (raw_csv->header[pos] == raw_csv->opts.delimiter ||
-					raw_csv->header[pos] == raw_csv->opts.terminator) {
-					raw_csv->col_names.emplace_back(raw_csv->header.data() + prev, pos - prev);
-					h_num_cols++;
-					if (raw_csv->header[pos] == raw_csv->opts.terminator) {
-						break;
-					}
-					prev = pos + 1;
-				}
-			}
-		} else {
-			// Assign column indexes as names if the header column is not present
-			h_num_cols = std::count(raw_csv->header.begin(), raw_csv->header.end(), raw_csv->opts.delimiter) + 1;
-			raw_csv->col_names.reserve(h_num_cols);
-			const string str_prefix = args->prefix != nullptr ? string(args->prefix) : "";
-			for (int i = 0; i < h_num_cols; ++i) {
-				raw_csv->col_names.push_back(str_prefix + std::to_string(i));
-			}
+
+		error = setColumnNamesFromCsv(raw_csv);
+		if (error != GDF_SUCCESS) {
+			return error;
 		}
+		const int h_num_cols = raw_csv->col_names.size();
 
 		// Allocating a boolean array that will use to state if a column needs to read or filtered.
 		raw_csv->h_parseCol = (bool*)malloc(sizeof(bool) * (h_num_cols));
 		RMM_TRY( RMM_ALLOC((void**)&raw_csv->d_parseCol,(sizeof(bool) * (h_num_cols)),0 ) );
 		for (int i = 0; i<h_num_cols; i++)
 			raw_csv->h_parseCol[i]=true;
+		
+		// Rename empty column names to "Unnamed: col_index"
+		for (size_t col_idx = 0; col_idx < raw_csv->col_names.size(); ++col_idx) {
```

Comment:
Question:  Did you add this to match pandas?

Reply:
Yes. Mostly the reason is that the output was pretty ugly with empty column names before this change; unnamed columns were treated as duplicates, so the names ended up as '', '.1', '.2'...